doc/local_context_guidelines.md

```diff
@@ -0,0 +1,139 @@
+The document provides instructions and guidelines on how to board local context feature for parameters.
+
+## Local context attribute
+
+Local context attribute is an abstraction of the definition for local context. We use this class to define how the argument support local context.
+
+There are 3 properties of local context attribute.
+
+- **name**
+
+`name` is the argument name used in local context. Usually just use the name which defined in the function signature, for example `resource_group_name`.
+
+- **scopes**
+
+`scopes` is an array of string. It defines where can this local context value been referenced. The value could be a command group or a command, for example `['vm', 'network']`. If one argument can be referenced in all the commands, you can define it as `['all']`. `scopes` is meaningful only when `SET` is in actions.
+    
+- **actions**
+
+`actions` is an array of [`LocalContextAction`](#Local context action). It defines whether set the command parameter value to local context or get value from local context. We can define both `GET` and `SET` action for one argument.
```

Comment:
It is kind of confusing
```
[`LocalContextAction`](#Local context action)
```

Reply:
This is a link, which will link to the definition for `LocalContextAction`